src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -1216,6 +1216,57 @@ public function testAutowireAttribute()
         $this->assertSame('@bar', $service->escapedRawValue);
         $this->assertNull($service->invalid);
     }
+    
+    public function testAutowireAttributeNullFallbackTestRequired()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register(AutowireAttributeNullFallback::class)
+            ->setAutowired(true)
+            ->setPublic(true)
+        ;
+
+        (new AutowirePass())->process($container);
+
+        $definition = $container->getDefinition(AutowireAttributeNullFallback::class);
+
+        $this->assertCount(2, $definition->getArguments());
+        $this->assertEquals(new Reference('required.parameter', ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE), $definition->getArgument(0));
+        $this->assertEquals(new Reference('optional.parameter', ContainerInterface::NULL_ON_INVALID_REFERENCE), $definition->getArgument(1));
+
+        $container->compile();
+
+        try {
+            $service = $container->get(AutowireAttributeNullFallback::class);
+        } catch (AutowiringFailedException $e) {
+            $this->assertSame('You have requested a non-existent parameter "required.parameter".', (string) $e->getMessage());
+        }
+    }
+        
+    public function testAutowireAttributeNullFallbackTestOptional()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register(AutowireAttributeNullFallback::class)
+            ->setAutowired(true)
+            ->setPublic(true)
+        ;
+        
+        $container->setParameter('required.parameter', 'foo');
+        
+        (new AutowirePass())->process($container);
+
+        $definition = $container->getDefinition(AutowireAttributeNullFallback::class);
+
+        $this->assertCount(2, $definition->getArguments());
+        $this->assertEquals(new Reference('required.parameter', ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE), $definition->getArgument(0));
+        $this->assertEquals(new Reference('optional.parameter', ContainerInterface::NULL_ON_INVALID_REFERENCE), $definition->getArgument(1));
```

Comment:
Same as above, might need to be `$this->assertEquals(null, $definition->getArgument(1));` ?